### PR TITLE
Validate email when creating or updating a mobile user

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

https://dimagi-dev.atlassian.net/browse/SAAS-12609

There is currently no email validation for the 'email' field on mobile users via the API. This ensures that when creating or updating a mobile user, the email is validated via django's email validator.

The email field is optional when creating a mobile user, which is why I chose to remove it from the initial create call, knowing it will be passed into the update method. I also noticed the errors returned from the `_update` method weren't actually being handled (maybe a reason to switch back to raising the first exception rather than a list of errors).

The theme of these improvements has been _incremental_, so if you see something inherent to the design that you would like to change but is out of this scope, feel free to create a ticket to improve in the future.

### Automated test coverage

Added a test to cover this.
### QA Plan

https://dimagi-dev.atlassian.net/browse/QA-4114 (will not run until post-deploy)

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by calellowitz — APPROVED** [review]

**Comment by gherceg** [comment]
I should have caught this the first time around as it was in the back of my mind. If the create request fails after the initial user object was created, we then delete the user object. This has the unfortunate side effect of rendering that username unusable on a followup request, which doesn't seem like great behavior. However the alternative isn't much better, which is to create the user, and then fail on specific invalid fields (like email), and expect the user to see that error and run an update method to fix it. I'm conflicted.

**Review by snopoke — APPROVED** [review]

File: corehq/apps/api/user_updates.py [hunk 1]
```diff
@@ -42,7 +43,12 @@ def update(user, field, value, user_change_logger=None):
 
 
 def _update_email(user, email, user_change_logger):
-    _simple_update(user, 'email', email.lower(), user_change_logger)
+    email = email.lower()
+    try:
+        validate_email(email)
+    except ValidationError:
+        raise UpdateUserException(_("The value '{}' for 'email' must be a valid email address").format(email))
```
Comment (joxl) [inline corehq/apps/api/user_updates.py:50]: Nit: (feel free to disregard) `the value <val> for 'email' must be a valid` feels a little awkward to me.
```suggestion
        raise UpdateUserException(_("Invalid 'email' field (must be a valid email address): {!r}").format(email))
```

**Review by joxl — APPROVED** [review]

**Comment by joxl** [comment]
> If the create request fails after the initial user object was created, we then delete the user object.

Naive question here, but how could the user object get successfully created with invalid property/field values? That seems like the core issue to me.

**Comment by gherceg** [comment]
Closing since I haven't touched this in a year

**Pull request closed without merging**